Assign commits to the standard GitHub Actions bot

per1234 opened the pull request:
The action and reusable workflow commits the generated documentation to the repository. A username and email address must be configured for this commit.

## Change Default Commit Author

Previously, [an arbitrary GitHub account](https://github.com/actions-user) was used for this purpose. It is questionable whether it is appropriate to attribute commits to an arbitrary user who didn't make them and which was not explicitly provided for such a purpose. In addition, doing so is problematic because we require all contributors to sign a CLA, and so we would need to set up an exemption for this arbitrary user in order to allow the commits to pass the [CLA signing check](https://github.com/cla-assistant/cla-assistant).

So the better approach is to assign the commit to the standard GitHub Actions bot account. This account is already assigned machine generated commits by the infrastructure of several of Arduino's repositories and is already exempted from the CLA check:

- https://github.com/arduino/arduino-ide/blob/6d96e227eb9471ce923fcad4400c26c27cc32ed1/.github/workflows/i18n-weekly-pull.yml#L60
- https://github.com/arduino/arduino-ide/blob/6d96e227eb9471ce923fcad4400c26c27cc32ed1/.github/workflows/themes-weekly-pull.yml#L69
- https://github.com/arduino/arduino-cli/blob/b9edb782a265b99878b4bce489c1751c4dcbee61/.github/workflows/i18n-weekly-pull.yaml#L50
- https://github.com/arduino/arc/blob/549f3eed627d571fb4d5f11d83020db68f264f75/.github/workflows/figma-style-weekly-pull.yaml#L41

For example:

https://github.com/arduino/arduino-ide/pull/2597/commits

![image](https://github.com/user-attachments/assets/2f0a7f9a-2658-4421-b637-1ed498a77650)

There is some evidence that this usage of the bot is expected by GitHub:

https://github.com/actions/checkout/blob/main/README.md#push-a-commit-using-the-built-in-token

(there is some interesting related information in https://github.com/actions/checkout/pull/1707)

---

An alternative would have been to use Arduino's general purpose @ArduinoBot account. This account is assigned automated commits by some infrastructure used exclusively in Arduino's own projects. However, the "arduino/render-docs-github-action" action and reusable workflow may also be used in projects not owned by Arduino, and it probably isn't appropriate to assign commits to ArduinoBot in those repositories.

## Make Commit Author Configurable

The users of the action and reusable workflow may wish to attribute commits for the rendered documentation to a specific user. This can now be done via the newly added `committer-name` and `committer-email` action inputs.

The inputs are optional, using the GitHub Actions bot account as the default author.

per1234 committed:
Assign commits to the standard GitHub Actions bot

The action and reusable workflow commits the generated documentation to the repository. A username and email address
must be configured for this commit.

Previously, an arbitrary GitHub account was used for this purpose. It is questionable whether it is appropriate to
attribute commits to an arbitrary user who didn't make them and which was not explicitly provided for such a purpose. In
addition, doing so is problematic because we require all contributors to sign a CLA, and so we would need to set up an
exemption for this arbitrary user in order to allow the commits to pass the CLA signing check.

So the better approach is to assign the commit to the standard GitHub Actions bot account. This account is already
assigned machine generated commits by the infrastructure of several of Arduino's repositories and is already exempted
from the CLA check.

An alternative would have been to use Arduino's general purpose "ArduinoBot" machine account. This account is assigned
automated commits by some infrastructure used exclusively in Arduino's own projects. However, the
"arduino/render-docs-github-action" action and reusable workflow may also be used in projects not owned by Arduino, and
it probably isn't appropriate to assign commits to ArduinoBot in those repositories.

sebromero commented:
@per1234 Thank you for your input! What if I make this a configurable parameter? Then we can configure the action to use `ArduinoBot` for Arduino repositories.

per1234 committed:
Make commit author configurable

The users of the action and reusable workflow may wish to attribute commits for the rendered documentation to a specific
user. This can now be done via the newly added `committer-name` and `committer-email` action inputs.

The inputs are optional, using the GitHub Actions bot account as the default author, as was done before the addition of
the inputs.

per1234 commented:
It sounds good @sebromero. I have added the inputs as you proposed:

[`a702b22` (#1)](https://github.com/arduino/render-docs-github-action/pull/1/commits/a702b225721b91667cb3ea1bf344842d898b6d20)

While I was working on that, I noticed that I had not updated the commit author data in the duplicated code for the reusable workflow. I have now corrected that omission so the reusable workflow and action are both updated.

sebromero commented on .github/workflows/render-docs.yml:
```diff
@@ -38,6 +38,14 @@ on:
+        type: string
+      committer-email:
+        description: 'Email address to use as author of the commit.'
+        default: '41898282+github-actions[bot]@users.noreply.github.com'
```
```suggestion
        default: 'github-actions[bot]@users.noreply.github.com'
```
I think I'd rather put a generic e-mail as default and set the Arduino specific one in the repositories that use this workflow. Alternatively we could remove the default value and make this a required parameter.

sebromero replied:
...ah, wait, so 41898282 is already a "generic" user ID?

per1234 replied:
It is the user ID for the "github-actions[bot]" user:

https://api.github.com/users/github-actions[bot]

Note this part:

```json
  "created_at": "2018-07-30T09:30:16Z",
```

The "noreply" email format is documented here:

https://docs.github.com/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address

> If you created your account after July 18, 2017, your noreply email address is an ID number and your username in the form of [ID+USERNAME@users.noreply.github.com](mailto:ID+USERNAME@users.noreply.github.com).

Since the "github-actions[bot]" account was created after 2017-07-18, the email address for the  user is `41898282+github-actions[bot]@users.noreply.github.com` (`41898282` is the user ID for github-actions[bot]).

sebromero reviewed (changes requested):
Please see comment

sebromero reviewed (approved):
LGTM
